mne/viz/evoked.py

```diff
@@ -1191,3 +1191,350 @@ def plot_evoked_joint(evoked, times="peaks", title='', picks=None,
+    else:
+        if isinstance(picks, int):
+            ch_names = [example.ch_names[picks]]
+            picks = [picks]
+        elif isinstance(picks, string_types):
+            ch_names = [picks]
+            picks = [example.ch_names.index(picks)]
+        elif isinstance(picks[0], int):
+            ch_names = [example.ch_names[pick] for pick in picks]
+        elif isinstance(picks[0], string_types):
+            ch_names = picks[:]
+            picks = [example.ch_names.index(pick) for pick in picks]
+        else:
+            raise ValueError("`picks` must be int, a list of int, "
+                             "str, a list of str, or None, not " +
+                             str(type(picks)))
+        ch_type = channel_type(example.info, picks[0])
+    scaling = _handle_default("scalings")[ch_type]
+
+    if ch_type == 'grad' and picks is not None:
+        from ..channels.layout import _merge_grad_data, _pair_grad_sensors
+        picked_chans = []
+        pairpicks = _pair_grad_sensors(example.info, topomap_coords=False)
+        for pick in picks:
+            picked_chans.append(pairpicks.index(pick))
+            picked_chans.append(pairpicks.index(pick - 1)
+                                if pick % 2 else pairpicks.index(pick + 1))
+        picks = list(sorted(set(picked_chans)))
+        ch_names = [example.ch_names[pick] for pick in
+                    [pairpicks[p_] for p_ in picks]]
+        if ymin is None:  # 'grad' is plotted as all-positive
+            ymin = 0
+
+    # deal with dict/list of lists, and calculate the CI
+    if not all([isinstance(evoked_, Evoked) for evoked_ in evokeds.values()]):
+        from ..evoked import combine_evoked
+        if not isinstance(evokeds[conditions[0]][0], Evoked):
+            raise ValueError("evokeds must be an `mne.Evoked` "
+                             "or of a collection of `mne.Evoked`s")
+
+        if ci is not None and picks is not None:
+            if not isinstance(ci, np.float):
+                msg = '`ci` must be float, got {0} instead.'
+                raise TypeError(msg.format(type(ci)))
+            # calculate the CI
+            sem_array = {}
+            for condition in conditions:
+                # this will fail if evokeds do not have the same structure
+                # (e.g. channel count)
+                if ch_type == 'grad' and picks is not None:
+                    data = np.asarray([
+                        _merge_grad_data(
+                            evoked_.data[pairpicks, :]).mean(0)[picks, :]
+                        for evoked_ in evokeds[condition]])
+                else:
+                    data = np.asarray([evoked_.data[picks, :].mean(0)
+                                       for evoked_ in evokeds[condition]])
+                sem_array[condition] = _ci(data, ci)
+
+        # get the grand mean
+        for cond in conditions:
+            evokeds[cond] = combine_evoked(evokeds[cond], weights='equal')
```

Comment:
>Because dictionaries are mutable this overwrites the contents of the original dictionary that the user passed

Candidate for the worst bug I've ever committed ...